sdk/core/azure-core/inc/azure/core/internal/io/filehandleholder.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief An internal FILE* Holder.
+ *
+ */
+#pragma once
+
+#include <cstdio>
+
+namespace Azure { namespace IO { namespace Internal {
+
+  /**
+   * @brief FileHandleHolder ensures native handle resource is released.
+   */
+  class FileHandleHolder {
```

Comment:
I wouldn't have this class at all.
Instead, I'd make your new FileBodyStream that takes a pathname string in the ctor call open/CreateFile, save the fd/handle in a field, and then call close/CloseHandle in the dtor.

I don't like the FILE* stream abstraction being dug into to grab the fd/handle, then saved in a field, and then closed. Aside from digging into an abstraction, the stream is holding onto memory that is not required to be held on to for the lifetime of the file being opened.